FirstApp/View/RecipeDetailViewController.swift

```diff
@@ -35,6 +39,44 @@ class RecipeDetailViewController: UIViewController {
         }
         self.navigationItem.title = navbar
         self.navigationController?.navigationBar.topItem?.title = sideNavBar
+        //swiftlint:disable all
+        database.collection("favourites").whereField("username", isEqualTo: username).whereField("title",
+                                                                                                 isEqualTo: recipeTitle)
+            .getDocuments() {(_, err) in
+            if let err = err {
+                print("Error getting documents: \(err)")
+            } else {
+                self.favouriteButton.setBackgroundImage(UIImage(systemName: "star.fill"), for: UIControl.State.normal)
+            }
+        }
+        //swiftlint:enable all
+    }
+
+    @IBAction func favouriteButtonTapped(_ sender: UIButton) {
+        print("Details: \(username)")
+        let currentBackgroundImage = favouriteButton.currentBackgroundImage
+        if currentBackgroundImage == UIImage(systemName: "star.fill") {
+            favouriteButton.setBackgroundImage(UIImage(systemName: "star"), for: UIControl.State.normal)
+            //swiftlint:disable all
+            database.collection("favourites").whereField("username", isEqualTo: username).whereField("title",
+                                                                                                     isEqualTo: recipeTitle)
+                .getDocuments() { (snapshot, err)
+                    in
+                if let err = err {
+                    print("Error getting documents: \(err)")
+                } else {
+                    snapshot?.documents.first?.reference.delete()
+                }
+            }
```

Comment:
I would advice that something like this be in its own function. Makes the code much more readable and if you need the code again you have it ready to use.

Reply:
Cool, will do🙂👍